src/Angular.js

```diff
@@ -1772,6 +1772,48 @@ function getTestability(rootElement) {
   return injector.get('$$testability');
 }
 
+function rootElementProviderFactory(rootElement) {
+  return ['$shutdownProvider', function($shutdownProvider) {
+    $shutdownProvider.register(function() {
+      if (rootElement.dealoc) {
+        rootElement.dealoc();
+      } else {
+        rootElement.find('*').removeData();
```

Comment:
No, it shouldn't. These are independent apps, so destroying the outer, shouldn't destroy the inner. To be clear, this is what I am talking about:

```html
<body app1>
  Hello from {{ 'app 1' }}!
  <div ng-non-bindable>
    <div app2>
      Hello from {{ 'app 2' }}!
    </div>
  </div>
</body>
```

```js
angular.module('app1', []);
angular.module('app2', []);

angular.bootstrap(document.querySelector('[app1]'), ['app1']);
angular.bootstrap(document.querySelector('[app2]'), ['app2']);
```